src/libs/tools/src/backendbuilder.cpp

```diff
@@ -409,6 +415,40 @@ void BackendBuilder::addPlugin (PluginSpec const & plugin)
 		newPlugin.appendConfig (provides.getConfig ());
 	}
 
+	// call plugin's checkconf function (if provided)
+	// this enables a plugin to verify its configuration at mount time
+	checkConfPtr checkConfFunction = nullptr;
+	try
+	{
+		checkConfFunction = reinterpret_cast<checkConfPtr> (pluginDatabase->getSymbol (newPlugin, "checkconf"));
+	}
+	catch (...)
+	{
+		// the checkconf operation is optional, so no worries if it was not found
+	}
+
+	if (checkConfFunction != nullptr)
+	{
+		ckdb::Key * errorKey = ckdb::keyNew (0);
+		ckdb::KeySet * pluginConfig = newPlugin.getConfig ().dup ();
+		int checkResult = checkConfFunction (errorKey, pluginConfig);
+		if (checkResult == -1)
+		{
+			ckdb::ksDel (pluginConfig);
+			throw PluginConfigInvalid (errorKey);
+		}
+		else if (checkResult == 1)
+		{
+			KeySet modifiedPluginConfig = KeySet (pluginConfig);
+			newPlugin.appendConfig (modifiedPluginConfig);
```

Comment:
It is easily possible to remove keys from a keyset within checkconf (simply use `clear()`). So I am against such a change of the `checkconf` signature.

But it is not possible with PluginSpec because you currently get only a `const KeySet`. So we would need a `setConfig` in the PluginSpec.

Additionally, it would be much better if you can also pass the backendConfig to the `checkconf`. The partial config might fail sometimes even though the whole config (backend+plugin) would pass.

Then we would also be able to use `checkconf` for the resolver. (`system/path` is added as backendConfig during the mount-process).